src/form-elements/FormElementCivicaStreetName.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react'
+import { formService } from '@oneblink/apps'
+
+import createTypedAutocompleteDropdown from '../components/AutocompleteDropdown'
+import FormElementLabelContainer from '../components/FormElementLabelContainer'
+import { FormTypes, CivicaTypes } from '@oneblink/types'
+
+const AutocompleteDropdown =
+  createTypedAutocompleteDropdown<CivicaTypes.CivicaStreetName>()
+type Props = {
+  formId: number
+  id: string
+  element: FormTypes.CivicaStreetNameElement
+  value: CivicaTypes.CivicaStreetName | undefined
+  displayValidationMessage: boolean
+  validationMessage: string | undefined
+  onChange: FormElementValueChangeHandler<CivicaTypes.CivicaStreetName>
+}
+
+function FormElementCivicaStreetName({
+  formId,
+  id,
+  element,
+  value,
+  displayValidationMessage,
+  validationMessage,
+  onChange,
+}: Props) {
+  const [label, setLabel] = React.useState('')
+  const [error, setError] = React.useState<Error | undefined>()
+
+  const handleSearch = React.useCallback(
+    async (searchTerm: string, abortSignal) => {
+      setError(undefined)
+      const results = await formService.searchCivicaStreetNames(
+        formId,
+        {
+          search: searchTerm,
+        },
+        abortSignal,
+      )
+      return results.map((suggestion, index) => ({
+        value: suggestion.formattedAccount || index.toString(),
+        label: suggestion.formattedStreet || index.toString(),
+        data: suggestion,
+      }))
+    },
+    [formId],
+  )
+
+  const handleChange = React.useCallback(
+    async (
+      streetId: string | undefined,
+      streetData: CivicaTypes.CivicaStreetName | undefined,
+    ) => {
+      if (!streetId || !streetData) {
+        onChange(element, undefined)
+        return
+      }
```

Comment:
this `if` redundant, the line below will either pass `streetData` or `undefined`